### ISSUE TITLE
Signed-ness of string

### ISSUE THREAD

**Issue by mpascoli** [yoppeh/efi-clang#2]
Hello, I noticed that I have to (CHAR16*)L"a string". Having to coerce all strings seemed to me a bit strange, so I wonder if there's a reason why you typedef'd CHAR16 to be a int16_t.
I checked in Tianocore's UDK and they #define _EFI_WCHAR_T      UINT16 (StdLib/Include/sys/EfiCdefs.h).
Also gnu-efi has: typedef UINT16 CHAR16; (efidef.h)

**Comment by yoppeh** [comment]
Yeah, I didn't look at those definitions, I went off of the spec, which doesn't really specify whether it should be signed or unsigned (that I saw). I think I just used a signed int since it didn't specifically say it had to be unsigned. I'll look into it. If it all checks out, I'll switch it to an uint16_t. But that change would be made [https://github.com/yoppeh/efi](https://github.com/yoppeh/efi), not this project.
